system/BaseModel.php

```diff
@@ -1916,4 +1888,24 @@ protected function convertToReturnType(array $row, string $returnType): array|ob
 
         return $this->converter->reconstruct($returnType, $row);
     }
+
+    /**
+     * Checking that the ID has a rational value.
```

Comment:
I would propose naming this method as `isIdAllowedAsPrimaryKey()` as that is really the purpose of this method. Or maybe `isIdValidAsPrimaryKey()` something.